### ISSUE TITLE
mockStatic throws java.lang.IllegalArgumentException: Cannot subclass final class 

### ISSUE THREAD

**Issue by johanhaleby** [jayway/powermock#229]
_From [yogeshpa...@gmail.com](https://code.google.com/u/101766227837128921499/) on November 30, 2009 12:10:25_

What steps will reproduce the problem? 1. Create a final class that has static methods
2. in setup method of the test class, create a static mock using mockStatic
3. run mvn test(assuming maven structure) What is the expected output? What do you see instead? Expected output : No test failures.
Actual output: 
java.lang.IllegalArgumentException: Cannot subclass final class...
        at net.sf.cglib.proxy.Enhancer.generateClass(Enhancer.java:442)
        at
net.sf.cglib.core.DefaultGeneratorStrategy.generate(DefaultGeneratorStrategy.java:25)
        at
net.sf.cglib.core.AbstractClassGenerator.create(AbstractClassGenerator.java:215)
        at net.sf.cglib.proxy.Enhancer.createHelper(Enhancer.java:373)
        at net.sf.cglib.proxy.Enhancer.createClass(Enhancer.java:313)
        at
org.powermock.api.easymock.internal.signedsupport.SignedSupportingClassProxyFactory.createProxy(SignedSupportingClassProxyFactory.java:155)
        at org.easymock.internal.MocksControl.createMock(MocksControl.java:40)
        at
org.powermock.api.easymock.PowerMock.doCreateMock(PowerMock.java:2155)
        at org.powermock.api.easymock.PowerMock.doMock(PowerMock.java:2110)
        at org.powermock.api.easymock.PowerMock.mockStatic(PowerMock.java:290)
j What version of the product are you using? On what operating system? Powermock 1.3.1 Please provide any additional information below.

_Original issue: http://code.google.com/p/powermock/issues/detail?id=209_

**Comment by johanhaleby** [comment]
_From [johan.ha...@gmail.com](https://code.google.com/u/105676376875942041029/) on November 30, 2009 07:22:25_

Have you really prepared the class for test using @PrepareForTest? Are you using the
PowerMockRunner? If you are, which class is it?

**Comment by johanhaleby** [comment]
_From [johan.ha...@gmail.com](https://code.google.com/u/105676376875942041029/) on December 03, 2009 04:33:54_

Marked as invalid because of no response

**Status:** Invalid

**Comment by praducg** [comment]
Even I am also facing the same issue.I am mocking interface array which throws java.lang.IllegalArgumentException: Cannot subclass final class class.
Following are the changes I did.
I did @PrepareForTest({ Array1[].class, Array2[].class })
inside the class I am doing like this
Array1[] crfngProcessor = PowerMockito.mock(Array1[].class);
Array2[] crfNgViews = PowerMockito.mock(Array2[].class);
and inside test method 
Mockito.when(staticclass.method()).thenReturn(crfngProcessor);

Basically I need to mock an array of interfaces.
Any help would be appreciated.
Thanks
Pradeep

**Comment by christopherbryanboyd** [comment]
I am encountering this with PowerMock and EasyMock, trying to mock java.util.ServiceLoader.

**Comment by prasadrevanaki** [comment]
I'm running junit test with @RunWith(PowerMockRunner.class) and @PrepareForTest({ AWSCognitoIdentityProviderClientBuilder.class}) at class level and mocking this class in setUp() method with 
PowerMockito.mockStatic(AWSCognitoIdentityProviderClientBuilder.class);
AWSCognitoIdentityProviderClientBuilder awsCognitoIdentityProviderClientBuilderMock = PowerMockito.mock(AWSCognitoIdentityProviderClientBuilder.class);

but I'm facing similar issue while building cognito client 

Cannot subclass final class class com.amazonaws.services.cognitoidp.AWSCognitoIdentityProviderClientBuilder
